golang/cosmos/e2e_test/pfm_test.go

```diff
@@ -0,0 +1,613 @@
+package e2etest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/agoric-labs/interchaintest/v6"
+	"github.com/agoric-labs/interchaintest/v6/chain/cosmos"
+	"github.com/agoric-labs/interchaintest/v6/ibc"
+	"github.com/agoric-labs/interchaintest/v6/testreporter"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestPFM(t *testing.T) {
+	cs := getChainSpec(t)
+	rf := getRelayerFactory(t)
+
+	testPFM(t, cs, rf)
+}
+
+// ForwardMetadata is the shape of the JSON in a single PFM hop in the IBC memo
+type ForwardMetadata struct {
+	Receiver       string        `json:"receiver"`
+	Port           string        `json:"port"`
+	Channel        string        `json:"channel"`
+	Timeout        time.Duration `json:"timeout"`
+	Retries        *uint8        `json:"retries,omitempty"`
+	Next           *string       `json:"next,omitempty"`
+	RefundSequence *uint64       `json:"refund_sequence,omitempty"`
+}
+
+// PacketMetadata is a chain of ForwardMetadata collectively describing the full IBC memo
+type PacketMetadata struct {
+	Forward *ForwardMetadata `json:"forward"`
+}
+
+// testPFM verified multiple PFM scenarios serially
+func testPFM(t *testing.T, cs []*interchaintest.ChainSpec, rf interchaintest.RelayerFactory) {
+	var (
+		ctx             = context.Background()
+		client, network = interchaintest.DockerSetup(t)
+		rep             = testreporter.NewNopReporter()
+		eRep            = rep.RelayerExecReporter(t)
+	)
```

Comment:
Just make these short variable declarations (e.g. `ctx := context.Background()`) unless there's some reason for the `var` block.

Reply:
They're only like that because I stole this test from the location referenced in the PR description and changed only what I needed to. Will follow up briefly in slack to get more opinion from you.